src/storage/core.rs

```diff
@@ -64,7 +65,7 @@ where
     pub(crate) blobs: Arc<RwLock<HierarchicalFilters<K, CombinedFilter<K>, Blob<K>>>>,
 }
 
-async fn work_dir_content(wd: &Path) -> Result<Option<Vec<DirEntry>>> {
+async fn work_dir_content(wd: &Path) -> Result<(Vec<DirEntry>, bool)> {
```

Comment:
I don't like this change. It is better to rewrite `count_old_corrupted_blobs` to not receive `[DirEntry]` as parameter